web/src/main/java/org/springframework/security/web/authentication/ServletAuthenticationManagerResolvers.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.authentication;
+
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.AuthenticationManagerResolver;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * A few simple factories for creating {@link AuthenticationManagerResolver}s based on various
+ * aspects of the {@link HttpServletRequest}
+ *
+ * @author Josh Cummings
+ * @since 5.2
+ * @see AuthenticationManagerResolver
+ */
+public final class ServletAuthenticationManagerResolvers {
```

Comment:
Let's rename this to AuthenticationManagerResolvers

Reply:
I'd like to also consider moving this to a class with a name more focused on multi tenancy. Please consider something like creating a class named `MultiTenancyAuthenticationManagerResolver` with static factory methods that do what is happening here. The `MultiTenancyAuthenticationManagerResolver` class allows injecting a strategy to resolve the tenant id from the request and then the manager from the tenant id.